Linking to arrow in a static x64-windows-static-md build incorrectly links ws2_32.dll.lib

**Describe the bug**
I'm trying to build my vcpkg as fully static ( less runtime libs, so x64-windows-static-md ).
Everything builds / links fine except one part of my project that links against arrow.  When linking Arrow I hit this error:
`LINK : fatal error LNK1181: cannot open input file 'ws2_32.dll.lib' [C:\Users\robam\csp\_skbuild\win-amd64-3.11\cmake-build\cpp\csp\python\adapters\parquetadapterimpl.vcxproj]`

Which looks odd on its face ( ".dll.lib" ? ).
When I looked a bit further I found this in the arrow cmake ( under vcpkg_installed\x64-windows-static-md\share\arrow\ArrowTargets.cmake:64 ):
`INTERFACE_LINK_LIBRARIES "OpenSSL::Crypto;OpenSSL::SSL;unofficial::brotli::brotlienc;unofficial::brotli::brotlidec;unofficial::brotli::brotlicommon;BZip2::BZip2;lz4::lz4;Snappy::snappy-static;ZLIB::ZLIB;zstd::libzstd_static;Threads::Threads;**ws2_32.dll**"`

Sure enough if I remove the .dll my build works fine.
I'm not sure if I'm doing something wrong or if that is some typo in the arrow cmake file ( forgive any ignorance on my part, its my first time using vcpkg )

**Environment**
- OS: Windows
- Compiler: MSVC 2019
- - Triplet: x64-windows-static-md

**To Reproduce**
Dont have an easy repro as my project is somewhat large, but if "ws2_32.dll" looks like its correct in that cmake file I can try to put one together.

**Expected behavior**
Since this is a static -md build I dont think ws2_32 should be placed with a .dll extension there )

--- a/ports/arrow/portfile.cmake
+++ b/ports/arrow/portfile.cmake
@@ -12,6 +12,7 @@ vcpkg_extract_source_archive(
         msvc-static-name.patch
         utf8proc.patch
         thrift.patch
+        remove-dll-suffix.patch #Upstream PR: https://github.com/apache/arrow/pull/41341
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/arrow/remove-dll-suffix.patch
+++ b/ports/arrow/remove-dll-suffix.patch
@@ -1,0 +1,39 @@
+diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
+index 941f5db..473e380 100644
+--- a/cpp/CMakeLists.txt
++++ b/cpp/CMakeLists.txt
+@@ -711,7 +711,7 @@ list(APPEND ARROW_TEST_LINK_LIBS ${ARROW_GTEST_GMOCK} ${ARROW_GTEST_GTEST_MAIN})
+ if(ARROW_BUILD_BENCHMARKS)
+   set(ARROW_BENCHMARK_LINK_LIBS benchmark::benchmark_main ${ARROW_TEST_LINK_LIBS})
+   if(WIN32)
+-    list(APPEND ARROW_BENCHMARK_LINK_LIBS Shlwapi.dll)
++    list(APPEND ARROW_BENCHMARK_LINK_LIBS Shlwapi)
+   endif()
+ endif()
+ 
+diff --git a/cpp/src/arrow/CMakeLists.txt b/cpp/src/arrow/CMakeLists.txt
+index 5c1b5e3..f16ee55 100644
+--- a/cpp/src/arrow/CMakeLists.txt
++++ b/cpp/src/arrow/CMakeLists.txt
+@@ -163,7 +163,7 @@ if(CMAKE_THREAD_LIBS_INIT)
+ endif()
+ 
+ if(WIN32)
+-  list(APPEND ARROW_SYSTEM_LINK_LIBS "ws2_32.dll")
++  list(APPEND ARROW_SYSTEM_LINK_LIBS "ws2_32")
+ endif()
+ 
+ if(NOT WIN32 AND NOT APPLE AND NOT ANDROID)
+@@ -628,9 +628,9 @@ else()
+   list(APPEND ARROW_TESTING_STATIC_INSTALL_INTERFACE_LIBS ArrowTesting::gtest)
+ endif()
+ if(WIN32)
+-  list(APPEND ARROW_TESTING_SHARED_LINK_LIBS "ws2_32.dll")
+-  list(APPEND ARROW_TESTING_STATIC_LINK_LIBS "ws2_32.dll")
+-  list(APPEND ARROW_TESTING_STATIC_INSTALL_INTERFACE_LIBS "ws2_32.dll")
++  list(APPEND ARROW_TESTING_SHARED_LINK_LIBS "ws2_32")
++  list(APPEND ARROW_TESTING_STATIC_LINK_LIBS "ws2_32")
++  list(APPEND ARROW_TESTING_STATIC_INSTALL_INTERFACE_LIBS "ws2_32")
+ endif()
+ 
+ set(ARROW_TESTING_SRCS

--- a/ports/arrow/vcpkg.json
+++ b/ports/arrow/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "arrow",
   "version": "16.0.0",
+  "port-version": 1,
   "description": "Cross-language development platform for in-memory analytics",
   "homepage": "https://arrow.apache.org",
   "license": "Apache-2.0",

--- a/versions/a-/arrow.json
+++ b/versions/a-/arrow.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "413da38f4ffabc5fd1273aff3ed2a7c6226f1ba6",
+      "version": "16.0.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "d68c109745e79bf3ae813428bd0cfda562790381",
       "version": "16.0.0",
       "port-version": 0

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -242,7 +242,7 @@
     },
     "arrow": {
       "baseline": "16.0.0",
-      "port-version": 0
+      "port-version": 1
     },
     "arsenalgear": {
       "baseline": "2.1.0",

